pkg/templates/reader.go

```diff
@@ -17,39 +17,33 @@ package templates
 import (
 	"fmt"
 
-	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
 	"sigs.k8s.io/controller-runtime/pkg/client"
 
 	"github.com/vmware-tanzu/cartographer/pkg/apis/v1alpha1"
-	"github.com/vmware-tanzu/cartographer/pkg/eval"
 )
 
-const PREVIEW_CHARACTER_LIMIT = 1024
+type Reader interface {
+	GetDefaultParams() v1alpha1.TemplateParams
 
-type Template interface {
+	// GetResourceTemplate returns the actual representation of a resource to stamp, and how to handle it
+	// TODO: we should be expecting something with a [ytt|template] interface, the health rules and params should
```

Comment:
The changes in pkg/templates/cluster_template.go seem to cut the opposite direction. Before we returned a struct with only the ytt and template field. Now a struct with those and the healthRules and params is returned.

Reply:
I'm not sure what you mean, we have removed a bunch of methods from this interface

Reply:
GetResourceTemplate used to be:
```go
func (t *clusterTemplate) GetResourceTemplate() v1alpha1.TemplateSpec {
	return v1alpha1.TemplateSpec{
		Template: t.template.Spec.Template,
		Ytt:      t.template.Spec.Ytt,
	}
}
```

and is now

```go
func (t *clusterTemplate) GetResourceTemplate() v1alpha1.TemplateSpec {
	return t.template.Spec
}
```

e.g. GetResourceTemplate is now returning a struct with healthRules and params included, which seems to run counter to your goals.

Reply:
No, we actually were making it consistent with the other templates and following the interface.